source/crud/bulk-write.md

```diff
@@ -410,17 +410,40 @@ required to implement. Drivers should follow the guidance in the CRUD specificat
 
 #### Summary vs. verbose results
 
+The `insertResults`, `updateResults`, and `deleteResults` fields MUST only be populated when `verboseResults` was set to
+true in the call to `MongoClient.bulkWrite`. Drivers MAY embed these results in a class rather than including them as
```

Comment:
I think this information is better suited for this section, so I simplified the language in the `hasVerboseResults` comment.

Reply:
To me the new state of affairs is even more confusing:

1. `BulkWriteResult` documents that when `BulkWriteResult.acknowledged` is `false`, _"all other members of this result will be undefined"_.
2. But it says nothing similar for a situation when `BulkWriteResult.hasVerboseResults` is `false` (currently, before the PR, there is no such inconsistency, `BulkWriteResult` speaks equally about both situations).
3. The new wording in "Summary vs. verbose results" is _"The `insertResults`, `updateResults`, and `deleteResults` fields MUST only be populated when `verboseResults` was set to true"_, while the wording about the state of these same fields when `BulkWriteResult.acknowledged` is `false` is _"will be undefined"_. The spec should use the same wording.

Reply:
> The spec should use the same wording.

I expect "populated" is used to mean a map is non-empty.

I expect "undefined" was copied from the CRUD spec [BulkWriteResult](https://github.com/mongodb/specifications/blob/c5dcae52a77d1ac90ca8c0cea7f371b73982062e/source/crud/crud.md#write-results). However, it is not clear to me what "undefined" means. I expect it may mean a field is not present (e.g. if the return type is a Javascript object) or maybe an error is thrown when accessing the property. Suggest clarifying what "undefined" means:

```typescript
/**
 * Indicates whether this write result was acknowledged. If not, drivers must prevent access to other fields.
 *
 * NOT REQUIRED TO IMPLEMENT. See below for more guidance on modeling unacknowledged results.
 */
acknowledged: Boolean;
```

Suggest adding a similar comment for the verbose results fields:

```typescript
/**
 * The results of each individual insert operation that was successfully performed.
 * 
 * NOT REQUIRED TO IMPLEMENT. See below for more guidance on modeling verbose results.
 */
insertResults: Map<Index, InsertOneResult>;
```

Reply:
Updated wording SGTM.